frontend/test/specs/comment/commentControllerSpec.js

```diff
@@ -0,0 +1,584 @@
+                    }
+                };
+            });
+            commentCtrl.replyComment();
+            expect(commentService.replyComment).toHaveBeenCalledWith(
+                post.key, reply.text, user.current_institution.key, comment.id
+            );
+            const replies = {};
+            replies[data.id] = data;
+            expect(commentCtrl.comment.replies).toEqual(replies);
+        });
+
+        it('should not create a new reply when newReply is null', function () {
+            commentCtrl.newReply = null;
+            expect(commentCtrl.comment.replies).toEqual({});
+            spyOn(commentService, 'replyComment').and.callFake(fakeCallback);
+            commentCtrl.replyComment();
+            expect(commentService.replyComment).not.toHaveBeenCalled();
+            expect(commentCtrl.comment.replies).toEqual({});
+        });
+    });
+
+    describe('deleteReply', function () {
+        it('should delete the reply', function () {
+            const replies = {};
+            commentCtrl.reply = {...reply};
+            replies[reply.id] = commentCtrl.reply;
+            replies['nil-key'] = nilReply;
+            commentCtrl.comment = {...comment , replies};
+            commentCtrl.setReplyId();
+            expect(commentCtrl.comment.replies).toEqual(replies);
+            spyOn(commentService, 'deleteReply').and.callFake(fakeCallback);
+            spyOn(messageService, 'showToast');
+            commentCtrl.deleteReply();
+            expect(commentService.deleteReply).toHaveBeenCalledWith(post.key, comment.id, reply.id);
+            expect(messageService.showToast).toHaveBeenCalledWith('Comentário excluído com sucesso');
+            expect(commentCtrl.comment.replies).toEqual({'nil-key': nilReply});
+        });
+    });
+
+    describe('deleteComment', function () {
+        it('should delete the comment', function () {
+            commentCtrl.post.data_comments = [comment];
+            expect(commentCtrl.post.data_comments).toEqual([comment]);
+            spyOn(commentService, 'deleteComment').and.callFake(fakeCallback);
+            spyOn(messageService, 'showToast');
+            commentCtrl.deleteComment();
+            expect(commentService.deleteComment).toHaveBeenCalledWith(post.key, comment.id);
+            expect(messageService.showToast).toHaveBeenCalledWith('Comentário excluído com sucesso');
+            expect(commentCtrl.post.data_comments).toEqual([]);
+        });
+    });
+
+    describe('confirmCommentDeletion()', function(){
+
+        it('Should delete the comment', function() {
+            spyOn(mdDialog, 'confirm').and.callThrough();
+            spyOn(mdDialog, 'show').and.callFake(fakeCallback);
+            spyOn(commentService, 'deleteComment').and.callThrough();
+            commentCtrl.post.data_comments = [comment];
+            httpBackend.expect('DELETE', POSTS_URI + '/' + post.key + '/comments/' + "5").respond(comment);
```

Comment:
you should put this http expect in the `deleteComent` test. 
And here only verify if the `deleteComent`function was called.